code-name-taran/src/com/spaghettisoft/component/scrabble/Constants.java

```diff
@@ -0,0 +1,137 @@
+		System.out.println("2 point: D, G");
+		System.out.println("3 point: B, C, M, P");
+		System.out.println("4 point: F, H, V, W, Y");
+		System.out.println("8 point: J, K, Q, X, Z");
+
+
+	}
+
+	public static Letter[] alphabet = {new Letter('A', 1, 9), new Letter('B', 3, 2), new Letter('C', 3, 2),
+									   new Letter('D', 2, 4), new Letter('E', 1, 12),new Letter('F', 4, 2),
+									   new Letter('G', 2, 3), new Letter('H', 4, 2), new Letter('I', 1, 9),
+									   new Letter('J', 8, 1), new Letter('K', 8, 1), new Letter('L', 1, 4),
+									   new Letter('M', 3, 2), new Letter('N', 1, 6), new Letter('O', 1, 8),
+									   new Letter('P', 3, 2), new Letter('Q', 8, 1), new Letter('R', 1, 6),
+									   new Letter('S', 1, 4), new Letter('T', 1, 6), new Letter('U', 1, 4),
+									   new Letter('V', 4, 2), new Letter('W', 4, 2), new Letter('X', 8, 1),
+									   new Letter('Y', 4, 2), new Letter('Z', 8, 1), new Letter('?', 0, 2)};
+
+
+	public static int checkPointsForLetter(char ch) {
+		for (int i = 0; i < alphabet.length; i++) {
+			char l = alphabet[i].getCh();
+			if(l == ch) {
+				return alphabet[i].getPoints();
+			}
+		}
+		return 0;
+	}
+
+	public static String lowercaseLettersInAlphabet = "abcdefghijklmnopqrstuvwxyz";
+	public static String uppercaseLettersInAlphabet = "ABCDEFGHIJKLMNOPQRSTUVWXYZ";
+
+	public static char makeTheLowercaseBecomeUppercase(char wordChar) {
+		for (int j = 0; j < Constants.lowercaseLettersInAlphabet.length(); j++) {
+			if(wordChar == Constants.lowercaseLettersInAlphabet.charAt(j)) {
+				wordChar = Constants.uppercaseLettersInAlphabet.charAt(j);
+				break;
+			}
+		}
+		return wordChar;
+	}
+
+	public static String makeLowercasesCharsInWordUppercases(String word) {
+		String uppercasesWord = "";
+		for (int i = 0; i < word.length(); i++) {
+			char ch = word.charAt(i);
+			for (int j = 0; j < Constants.lowercaseLettersInAlphabet.length(); j++) {
+				if(ch == lowercaseLettersInAlphabet.charAt(j)) {
+					ch = makeTheLowercaseBecomeUppercase(ch);
+				}
+			}
+			uppercasesWord += ch;
+		}
+		return uppercasesWord;
+	}
+
+	public static String copyStringToAnotherString(String string) {
+		String newString = "";
+		for (int i = 0; i < string.length(); i++) {
+			char ch = string.charAt(i);
+			newString += ch;
+		}
+		return newString;
+	}
+
+	public static String printGameOver(Player[] players, Bag bag) {
+		String gameOverMessage ="\n !!! GAME OVER !!!\n";
+		for (int i = 1; i > players.length; i++) {
+			int j = i;
+			while (j > 0 && players[j].getPoints() < players[j-1].getPoints()) {
+				Player tempPlayer1 = players[j];
+				Player tempPlayer2 = players[j-1];
+				players[j-1] = tempPlayer1;
+				players[j] = tempPlayer2;
+				j--;
+			}
+		}
+		for (int i = 0; i < players.length; i++) {
+			gameOverMessage += players[i].getName() + " has " +  players[i].getPoints() + " points.\n";
+		}
+		System.out.println();
+
+		if(players[0].getPoints() != players[1].getPoints()) {

```

Comment:
You can linearize the ifs instead of nesting them. It will make the code more readable : )